helmfile.d/charts/prometheus-alerts/templates/alerts/kubernetes-system-kubelet.yaml

```
@@ -176,28 +176,85 @@ spec:
 {{- if .Values.defaultRules.additionalRuleLabels }}
 {{ toYaml .Values.defaultRules.additionalRuleLabels | indent 8 }}
 {{- end }}
-    - alert: LessKubeletsThanNodes
+{{- if .Values.autoscaledNodeGroupAlerts.enabled }}
+    - alert: KubletDownForAutoscaledNodeFor15m
       annotations:
-        description: There are less kubelets than nodes for the last 5m.
-        runbook_url: {{ .Values.defaultRules.runbookUrl }}kubernetes/LessKubeletsThanNodes
-        summary: There are less kubelets than nodes for the last 5m.
-      expr: up{job="kubelet", metrics_path="/metrics"} == 0
+        description: The kubelet job of an existing autoscaled node {{`{{ $labels.node }}`}} in cluster {{`{{ $labels.cluster }}`}} is down for the last 15m.
+        summary: The kubelet job of an existing autoscaled cluster {{`{{ $labels.node }}`}} in cluster {{`{{ $labels.node }}`}} is down for the last 15m.
+      expr: |-
+          sum by (node, cluster) (kube_node_info and on (node) up{job="kubelet",metrics_path="/metrics"} == 0)
+          and on (node)
+          kube_node_labels{ {{ .Values.autoscaledNodeGroupAlerts.groupLabel }}{{ .Values.autoscaledNodeGroupAlerts.groupLabelValue.regex }} }
```

Suggestion: I feel like it's easier to understand how this works if you move the `=~` here

```suggestion
          kube_node_labels{ {{ .Values.autoscaledNodeGroupAlerts.groupLabel }}=~{{ .Values.autoscaledNodeGroupAlerts.groupLabelValue.regex }} }
```

Since we know it's a regex given the name of the variable.
But I could also accept leaving it like this if you feel differently

---

The problem is that we have do use `!=` when the groupLableValue is an empty array, that would add another if statement in the alert template, makes it a bit messy. So I choose to put the if statement in the value file.

---

Hmm, right. Yea that make sense, yea then keep it :)
Side note, isn't it `!~` when you're dealing with "not equals regex"?

> It is also possible to negatively match a label value, or to match label values against regular expressions. The following label matching operators exist:
> * `=`: Select labels that are exactly equal to the provided string.
> * `!=`: Select labels that are not equal to the provided string.
> * `=~`: Select labels that regex-match the provided string.
> * `!~`: Select labels that do not regex-match the provided string.

---

Yes, not equal is only used when the label value is empty, in this case we filter out all nodes with autoscaledNode label regardless the label value. It's always an empty string at right of this equation, not a regex expression.

---

@Xartos  Thought through about it, my previous explanation is correct. `!=` is only used when label value array is empty, so there won't be any regex expression on the right side of the expression. So there's no point using `!~`.
Here in the value file is how I decide when to use `=~` and when to use `!=`. 
```
autoscaledNodeGroupAlerts:
  enabled: {{ .Values.prometheus.autoscaledNodeGroupAlerts.enabled }}
  groupLabel: {{ $groupLabel }}
  {{- if $groupLabelValueArrayRegex }}
  groupLabelValue:
    regex: '=~"{{ $groupLabelValueArrayRegex }}"'
  {{- else }}
  groupLabelValue:
    regex: '!="{{ $groupLabelValueArrayRegex }}"'
  {{- end }}

```
The reason why `!=` is used here for filtering out nodes is that when the value array is empty, `kube_node_labels{groupLable=~""}` couldn't split nodes with and without `groupLable` label,  it returns the metrics for all nodes, so I had to use `kube_node_labels{groupLable!=""}` to separate them. When the value array is not empty, `kube_node_labels{groupLable=~"value1|value2"}` can split the nodes based on this label without any issue.

---

Riiight! Then maybe you could change it into

```
  {{- if $groupLabelValueArrayRegex }}
  groupLabelValue:
    regex: '=~"{{ $groupLabelValueArrayRegex }}"'
  {{- else }}
  groupLabelValue:
    regex: '!=""'
  {{- end }}
```

Just to make this a bit clearer? What do you think?

---

Yes that is more clear, will change that